src/main/java/seedu/address/model/person/patient/Patient.java

```diff
@@ -0,0 +1,120 @@
+package seedu.address.model.person.patient;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.tag.Tag;
+/**
+ * Represents a Patient in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Patient extends Person {
+    // Identity fields
+    private final Height height;
+    private final Weight weight;
+    private final Diagnosis diagnosis;
+    private final Status status;
+    private final Remark remark;
+    /**
+     * Every field must be present and not null.
+     */
+    public Patient(Name name, Phone phone, Email email, Height height, Weight weight,
+                   Diagnosis diagnosis, Status status, Remark remark, Set<Tag> tags) {
+        super(name, phone, email, tags);
+        requireAllNonNull(name, phone, email, height, weight, diagnosis, status, remark, tags);
```

Comment:
Do we need to check `requireAllNonNull` here for `name, phone, email` and `tags` if `super(name, phone, email, tags)` will do that for us?